src/main/java/seedu/duke/operationlist/PassengerList.java

```diff
@@ -0,0 +1,437 @@
+            validatePassenger(i);
+            if (isPassengerDuplicate()) {
+                resetCheckDuplicate();
+                throw new SkyControlException(ui.getDuplicatePassengerError());
+            }
+        }
+    }
+
+    private void modifyPassengersFlightNum(String flightNum, String newFlightNum) {
+        getNumberOfPassengers();
+        for (int i = 0; i < numOfPassengers; i++) {
+            PassengerInfo passenger = passengers.get(i);
+            if (passenger.getFlightNumber().equals(flightNum)) {
+                passenger.setFlightNumber(newFlightNum);
+            }
+        }
+    }
+
+    private void modifyPassengersGateNum(String flightNum, String newGateNum) {
+        getNumberOfPassengers();
+        for (int i = 0; i < numOfPassengers; i++) {
+            PassengerInfo passenger = passengers.get(i);
+            if (passenger.getFlightNumber().equals(flightNum)) {
+                passenger.setGateNumber(newGateNum);
+            }
+        }
+    }
+
+    //@@author ivanthengwr
+    @Override
+    public void deleteOperation(String passengerDetail) throws SkyControlException {
+        getNumberOfPassengers();
+        getPassengerName(passengerDetail);
+        getFlightNumber(passengerDetail);
+        getSeatNumber(passengerDetail);
+        for (int i = 0; i < numOfPassengers; i++) {
+            validatePassenger(i);
+            if (isValidPassenger) {
+                passengers.remove(i);
+                getNumberOfPassengers();
+                Ui.showDeleteMessage(name, flightNumber, seatNumber, numOfPassengers);
+                isDeleteSuccess = true;
+                break;
+            }
+        }
+        if (!isDeleteSuccess) {
+            throw new SkyControlException(ui.getDeleteError());
+        }
+        isDeleteSuccess = false;
+    }
+
+    @Override
+    public void listOperation() {
+        getNumberOfPassengers();
+        ui.showPassengerListHeader();
+        if (numOfPassengers == NO_PASSENGER) {
+            ui.showEmptyPassengerList();
+        }
+        for (int i = 0; i < numOfPassengers; i++) {
+            iteratePassengerDetail(i);
+            ui.showPassengerListBody(name, departureDate, departureTime,
+                    flightNumber, gateNumber, boardingGroup, seatNumber, boardingTime);
+        }
+        System.out.print("\n");
+    }
+
+    private void validatePassenger(int index) {
+        getNumberOfPassengers();
+        assert index < numOfPassengers;
+        isNamePresent = passengers.get(index).getName().contains(name);
+        isFlightNumberPresent = passengers.get(index).getFlightNumber().contains(flightNumber);
+        isSeatNumberPresent = passengers.get(index).getSeatNumber().contains(seatNumber);
+        isDepartureTimePresent = passengers.get(index).getDepartureTime().contains(departureTime);
+        isValidPassenger = isNamePresent && isFlightNumberPresent && isSeatNumberPresent && isDepartureTimePresent;
```

Comment:
Avoid complicated expressions, consider breaking them up into different steps or refactor into another method. This applies to the rest of the code as well.